contracts/managers/crowdfunding-softcap-vesting/CfManagerSoftcapVesting.sol

```diff
@@ -48,57 +48,80 @@ contract CfManagerSoftcapVesting is ICfManagerSoftcapVesting, ACfManager {
     //------------------------
     //  CONSTRUCTOR
     //------------------------
-    constructor(
-        string memory contractFlavor,
-        string memory contractVersion,
-        address owner,
-        address asset,
-        uint256 tokenPrice,
-        uint256 softCap,
-        uint256 minInvestment,
-        uint256 maxInvestment,
-        bool whitelistRequired,
-        string memory info,
-        address feeManager
-    ) {
-        require(owner != address(0), "CfManagerSoftcapVesting: Invalid owner address");
-        require(asset != address(0), "CfManagerSoftcapVesting: Invalid asset address");
-        require(tokenPrice > 0, "CfManagerSoftcapVesting: Initial price per token must be greater than 0.");
-        require(maxInvestment >= minInvestment, "CfManagerSoftcapVesting: Max has to be bigger than min investment.");
-        require(maxInvestment > 0, "CfManagerSoftcapVesting: Max investment has to be bigger than 0.");
-        IIssuerCommon issuer = IIssuerCommon(IAssetCommon(asset).commonState().issuer);
+    constructor(Structs.CampaignConstructor memory params) {
+        require(params.owner != address(0), "CfManagerSoftcapVesting: Invalid owner address");
+        require(params.asset != address(0), "CfManagerSoftcapVesting: Invalid asset address");
+        require(params.tokenPrice > 0, "CfManagerSoftcapVesting: Initial price per token must be greater than 0.");
+        require(
+            params.maxInvestment >= params.minInvestment,
+            "CfManagerSoftcapVesting: Max has to be bigger than min investment."
+        );
+        require(
+            params.maxInvestment > 0,
+            "CfManagerSoftcapVesting: Max investment has to be bigger than 0."
+        );
+
+        address fetchedIssuer = _safe_issuer_fetch(params.asset);
+        address issuerProcessed = fetchedIssuer != address(0) ? fetchedIssuer : params.issuer;
+        require(issuerProcessed != address(0), "CfManagerSoftcapVesting: Invalid issuer.");
```

Comment:
This could potentially cause problems because the user can define issuer in params but another issuer is set in the constructor. Maybe add `require` to this logic.